src/react-components/src/components/01-atoms/Images/Icons/Icon.tsx

```diff
@@ -0,0 +1,46 @@
+
+import * as React from "react";
+import Icons from "@nypl/design-system-icons/dist/img/sprite/sprite.svg";
+import bem from "../../../../utils/bem";
+
+  // Wrapper Class for Icon
+  export interface IconProps {
+    name: string;
+    blockName: string;
+    modifiers?: string[];
+    decorative: boolean;
+    role?: string;
+    title?: string;
+    desc?: string;
+  }
+
+  export default class Icon extends React.Component<IconProps, {}> {
+    constructor(props: IconProps) {
+      super(props);
+    }
+
+    render(): JSX.Element {
+      let { modifiers, blockName, name, decorative, role, title, desc } = this.props;
+      let icon_base_class = "icon";
+
+      let iconProps = {
+        "className": bem(icon_base_class, modifiers, blockName),
+        "role": decorative ? "img" : role,
+        "aria-hidden": decorative,
+        "aria-labelledby": title ? "title-" + name : undefined,
+        "aria-describedby": desc ? "desc-" + name : undefined
+      };
+
+      return (
+        <svg {...iconProps} >
+            {title &&
+              <title id={`title-${name}`}>{ title }</title>
+            }
+            {desc &&
+              <desc id={`title-${name}`}>{ desc }</desc>
+            }
+            <use xlinkHref={`${Icons}#${name}`} />
```

Comment:
I think the `xlinkHref` is deprecated and not recommend, no? The idea for this is that you'd declare an SVG icon on the page and reuse it whenever this is instantiated, right? If so, how will that first SVG be loaded?

Reply:
Yes, it is.  The way that this was done in Twig/Drupal was that all the svgs get compiled in one file and referenced by their ID (https://una.im/svg-icons/) and that's what SVGSpriter is doing. I'm not sure if it's tied to xlinkhref - I'll investigate that, I'm guessing I could just replace with href and it would be ok.

I don't actually know how they're being loaded, though.  I'm not ... actually sure the svg gets reused?

Reply:
I've come to like the solution proposed above, but it looks like the library that Twig is using hasn't been updated since 2016, which could be the reason it's producing `xlinkHref`s instead of `Hrefs` which is what MDN suggests you update to.

Messing with the icons in Patternlab, it looks like changing the `xlinkHref` to `href` doesn't have any adverse affects that I'm seeing in Chrome.

Reply:
So just taking changing `xlinkHref` to `href` seems to work out here.  It looks like it's being cached and re-used, too, I wonder if the import is doing that work for me.